newtest/reloader_test.go

```diff
@@ -0,0 +1,255 @@
+	}
+
+	if err := client.Post().
+		Resource("addons").
+		Namespace("kube-system").
+		Body("testdata/reloader.json").
+		Do(context.Background()).Error(); err != nil {
+		return fmt.Errorf("couldn't create reloader addon: %w", err)
+	}
+
+	return nil
+}
+
+func deployNginx(client kubernetes.Interface) (*corev1.ConfigMap, error) {
+	cm := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "html",
+		},
+		Data: map[string]string{
+			"index.html": `original`,
+		},
+	}
+	_, err := client.CoreV1().ConfigMaps("default").Create(context.Background(), &cm, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not create nginx ConfigMap: %w", err)
+	}
+
+	deployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "nginx",
+			Annotations: map[string]string{
+				"reloader.stakater.com/auto": "true",
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "nginx",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "nginx",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Name:  "nginx",
+						Image: "nginx:1.19.4",
+						VolumeMounts: []corev1.VolumeMount{{
+							Name:      "html",
+							MountPath: "/usr/share/nginx/html",
+							ReadOnly:  true,
+						}},
+					}},
+					Volumes: []corev1.Volume{{
+						Name: "html",
+						VolumeSource: corev1.VolumeSource{
+							ConfigMap: &corev1.ConfigMapVolumeSource{
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: "html",
+								},
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
+	_, err = client.AppsV1().Deployments("default").Create(context.Background(), &deployment, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not create nginx Deployment: %w", err)
+	}
+
+	if err := wait.Poll(5*time.Second, 60*time.Second, func() (bool, error) {
+		deploy, err := client.AppsV1().Deployments("default").Get(context.Background(), "nginx", metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return deploy.Status.ReadyReplicas > 0, nil
+	}); err != nil {
+		return nil, fmt.Errorf("couldn't check deployment readiness: %w", err)
+	}
+
+	_, err = client.CoreV1().Services("default").Create(context.Background(), &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "nginx",
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app": "nginx",
+			},
+			Ports: []corev1.ServicePort{{
+				Port: 80,
+			}},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not create nginx Service: %w", err)
+	}
+
+	return &cm, nil
+}
+
+func curl(c typedcorev1.PodInterface, url string) (string, error) {
+	pod, err := c.Create(context.Background(), &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "debug",
```

Comment:
It would be nice to have this defined as a constant with a clear name referring to its use. Reading later 

```
	defer func() {
		c.Delete(context.Background(), "debug", metav1.DeleteOptions{})
	}()
```

Wasn't clear to me what is happening.